### PR TITLE
Add FreeBSD CI checks

### PR DESCRIPTION
**Pull request by Neilpang**
Checks for :  FreeBSD  12.3, 13.0 and 13.1
For both cmake and auto tools.

### CODE REVIEW

**Review by azat — APPROVED** [review]

**Comment by Neilpang** [comment]
seems like a temp rsync error.  please rerun the failed tests again.
Thanks

**Comment by azat** [comment]
@Neilpang thank you for the effort, this is astonishing!

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -739,3 +739,177 @@ jobs:
...
+        uses: vmactions/freebsd-vm@v0
+        with:
+          release: ${{ matrix.release }}
+          prepare: |
+            pkg install -y  mbedtls  python3 autoconf automake libtool pkgconf
+          usesh: true
+          run: |
+            if [ "${{ matrix.EVENT_MATRIX }}" == "DISABLE_OPENSSL" ]; then
+              EVENT_CONFIGURE_OPTIONS="--disable-openssl"
+
```
Comment (azat) [inline .github/workflows/build.yml:872]: Egh, trailing whitespaces...

**Comment by azat** [comment]
I'm seeing that [sometimes it failes on rsync step](https://github.com/libevent/libevent/runs/7822356599?check_suite_focus=true), can this be retried somehow? But I guess that machine is down and the whole job should be retried?

```
  ssh_dispatch_run_fatal: Connection to 127.0.0.1 port 2222: message authentication code incorrect
  rsync: connection unexpectedly closed (485338 bytes received so far) [receiver]
  rsync error: error in rsync protocol data stream (code 12) at /AppleInternal/Library/BuildRoots/20d6c351-ee94-11ec-bcaf-7247572f23b4/Library/Caches/com.apple.xbs/Sources/rsync/rsync/io.c(453) [receiver=2.6.9]
  rsync: connection unexpectedly closed (34215 bytes received so far) [generator]
  rsync error: unexplained error (code 255) at /AppleInternal/Library/BuildRoots/20d6c351-ee94-11ec-bcaf-7247572f23b4/Library/Caches/com.apple.xbs/Sources/rsync/rsync/io.c(453) [generator=2.6.9]
  Error: The process '/bin/bash' failed with exit code 255
```

**Comment by Neilpang** [comment]
Yes,   the whole job should be retried.

**Comment by azat** [comment]
>Yes, the whole job should be retried.

I will be very grateful if you can take a look at this, since it now freebsd CI is flaky - https://github.com/libevent/libevent/runs/8057614577?check_suite_focus=true

**Comment by Neilpang** [comment]
PR  created: https://github.com/libevent/libevent/pull/1328

please check.

**Comment by azat** [comment]
https://github.com/libevent/libevent/commit/b9e1fe74e724a3f1524904b4658b9a4acdea50f5

**Comment by azat** [comment]
FYI I disabled them due to lack of runners - https://github.com/libevent/libevent/commit/b9e1fe74e724a3f1524904b4658b9a4acdea50f5

**Comment by Neilpang** [comment]
I just made a new pr to upgrade the vmactions to the latest version, which is using the linxu runners.
It's better and more stable.  please give it a try.

https://github.com/libevent/libevent/pull/1570

**Pull request merged**